scripts/fuzz_opt.py

```diff
@@ -1757,12 +1757,15 @@ def handle(self, wasm):
         # to the unprocessed original text.
         original = run([in_bin('wasm-opt'), wasm] + FEATURE_OPTS + ['--print'])
 
-        # We cannot run if the module has (ref exn) in globals (because we have
-        # no way to generate an exn in a non-function context). The fuzzer is
+        # We leave if the module has (ref exn) in struct fields (because we have
+        # no way to generate an exn in a non-function context, and if we picked
+        # that struct for a global, we'd end up needing a (ref enx) in the
```

Comment:
```suggestion
        # that struct for a global, we'd end up needing a (ref exn) in the
```
Do we also need to check for globals that have type `(ref exn)` like the previous version of this code did, or do we think that that should never occur?

Reply:
I don't think it can be a problem. The fuzzer doesn't create such things originally, and no optimization pass can either AFAICT. (Well, a pass can refine `(global $g (ref null exn) (..))` to a non-nullable type, but the value already existed there - so the value must still be valid to use.)

Reply:
Struct types are a hazard because (1) we can refine their fields to `(ref exn)` and also (2) we may decide to create new globals with them. A single existing global, in comparison, is fine.

Another option for structs could be to avoid creating globals for structs with such fields, but we'd need to recurse to the types of fields etc.; the current hack seems simpler.

Reply:
We do already have some logic for detecting uninhabitable heap types, e.g. those with cycles of non-nullable references. Perhaps we could extend it to detect types that are uninhabitable in constant expressions without adding new imports.

But the hack seems fine for now.